### ISSUE TITLE
padding_idx doesn't work.

### ISSUE THREAD

**Issue by TropComplique** [pytorch/pytorch#1812]
Hi. There is no zero padding:
```
import torch
from torch.autograd import Variable
import torch.nn as nn

e = nn.Embedding(3, 2, padding_idx=0)
e.weight = nn.Parameter(torch.rand(3, 2))
print(e.weight)
```
**Output:**
Parameter containing:
 0.5051  0.5076
 0.9464  0.2811
 0.4968  0.2109
[torch.FloatTensor of size 3x2]
```
indices = Variable(torch.LongTensor([0, 1, 2]))
e(indices)
```
**Output:**
Variable containing:
 0.5051  0.5076
 0.9464  0.2811
 0.4968  0.2109
[torch.FloatTensor of size 3x2]

**pytorch version**: 0.1.12

**Comment by ChijunZhang** [comment]
Remove  this line: e.weight = nn.Parameter(torch.rand(3, 2))
From nn.Embedding source, you could see:
 if self.padding_idx is not None:
    self.weight.data[self.padding_idx].fill_(0)

**Comment by TropComplique** [comment]
Yeah, it worked.
But it would be nice to be able to initialize embedding by hand.
So, if I do like this:
```
e = nn.Embedding(3, 2, padding_idx=0)
e.weight = nn.Parameter(torch.cat([torch.zeros(1, 2), torch.rand(2, 2)], dim=0))
```
Zero row will be kept constant during training?

**Comment by ChijunZhang** [comment]
I haven't run your code,but I believe it will be kept zeros for first row cause it`s backward gradient will always be zeros

**Comment by ChijunZhang** [comment]
I have tested your code, backward gradients for first row always be zeros.

**Comment by TropComplique** [comment]
@ChijunZhang Thank you for your help.
